Cython/Utility/ModuleSetupCode.c

```diff
@@ -1188,11 +1188,16 @@ static CYTHON_INLINE int __Pyx_PyDict_GetItemRef(PyObject *dict, PyObject *key,
 #endif
 
 // "Py_am_send" requires Py3.10 when using type specs (which utility code types do).
-#define __PYX_HAS_PY_AM_SEND  CYTHON_USE_AM_SEND && __PYX_LIMITED_VERSION_HEX >= 0x030A0000
+#if !CYTHON_USE_AM_SEND
+#define __PYX_HAS_PY_AM_SEND 0
+#elif __PYX_LIMITED_VERSION_HEX >= 0x030A0000
+#define __PYX_HAS_PY_AM_SEND 1
+#else
+#define __PYX_HAS_PY_AM_SEND 2  // our own backported implementation
+#endif
```

Comment:
IIUC, this means that a Cython module built for Limited API 3.9 is not compatible with one built for Limited API 3.10, right? (Assuming they use generators/coroutines.)

Reply:
"Not compatible" in the sense that they shouldn't share the same ABI module.

They don't currently have a separate ABI prefix and probably should

Based on https://github.com/cython/cython/blob/bf4e9a3a7b80f80eead0ee5621c74a0807b9a57e/Cython/Utility/ModuleSetupCode.c#L1237-L1245

I think the incompatibility predates this PR, but it's probably the time to fix it.

Reply:
I've added a different ABI suffix for this.

Thinking about it more, I'm not convinced it's _really_ a problem - I think all that will happen is that the slot won't be set and set won't get used. But it's a little hard to be absolutely sure so I've added the suffix.